metrics/web/lib/base/presentation/widgets/base_popup.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:flutter/material.dart';
+
+/// Signature for the function that builds an offset using the [childSize].
+typedef OffsetBuilder = Offset Function(Size childSize);
+
+/// A widget that displays the given [child] that opens the given [popup]
+/// when is triggered.
+class BasePopup extends StatelessWidget {
+  /// A widget to display as a [popup] trigger.
+  final Widget child;
```

Comment:
Should we really have a child in the popup widget that triggers the Popup? Maybe popup should be just popup without triggering mechanism? Or maybe improve naming for this to make it a little bit clear? Would love to hear your thoughts on this via chat.